### PR TITLE
171350919 multilang dictionaries

### PR DESCRIPTION
**Pull request by zaibacu**
**Work In Progress**

- Move current `yaml` dictionary files into multilanguage `edn` files

- In this PR we take out default values from metadata structure and use as phrases before. In Verbs: `(tense :present)`, in Nouns `(inflection :nom-sg)`

Word `place` now has `en` ir `de`, screenshot of simply adding single dict item into document plan and rendering in both languages:
basic-multilang

### CODE REVIEW

File: api/src/api/graphql/domain/dictionary.clj [hunk 1]
```diff
@@ -6,20 +6,34 @@
             [com.walmartlabs.lacinia.resolve :refer [resolve-as]]
             [data.entities.dictionary :as dict-entity]))
 
+;; (defn dictionary [_ _ _]
```
Comment (zmedelis) [inline api/src/api/graphql/domain/dictionary.clj]: Remove commented out code

File: api/src/api/graphql/translate/dictionary.clj [hunk 2]
```diff
@@ -38,3 +39,70 @@
                         :label  ""
                         :roles  []
                         :frames []}))}))
+
+(defn pos->schema [pos]
```
Comment (zmedelis) [inline api/src/api/graphql/translate/dictionary.clj]: Lets stick to POS naming as it is used in GF. See `morphology` schema for supported values

File: api/src/api/graphql/translate/dictionary.clj [hunk 3]
```diff
@@ -38,3 +39,70 @@
...
+    :v2  "VB"
+    :sconj "IN"
+    (name pos)))
+
+(defn multilang-dict-item->schema [{:keys [id key pos gender language senses inflections tenses] :as dict-item}]
+  (log/debugf "MultilangDictItem: %s" dict-item)
+  {:id            id
+   :key           key
+   :pos           (pos->schema pos)
+   :language      (name language)
```
Comment (zmedelis) [inline api/src/api/graphql/translate/dictionary.clj]: Review all uses of language and lets stick to GF use of them, that is a string of three letters "Eng", "Ger", "Lav", "Rus", "Est"

File: api/src/api/graphql/translate/dictionary.clj [hunk 4]
```diff
@@ -38,3 +39,70 @@
...
+  (let [value-map (into {} (map (fn [tense] {(:tense/key tense) (:tense/value tense)}) tenses))]
+    (get value-map :present "")))
+
+(defn get-default-inflection [inflections]
+  (let [value-map (into {} (map (fn [infl] {(:inflection/key infl) (:inflection/value infl)}) inflections))]
+    (get value-map :nom-sg "")))
+
+
+(defn multilang-dict-item->original-schema [{:keys [key pos]} items]
+  (let [lang-translation {"English"    :en
```
Comment (zmedelis) [inline api/src/api/graphql/translate/dictionary.clj]: Note as per previous comment this map will be "English" -> "Eng" etc

File: api/src/api/graphql/translate/dictionary.clj [hunk 5]
```diff
@@ -38,3 +39,70 @@
...
+  {:id            id
+   :key           key
+   :pos           (pos->schema pos)
+   :language      (name language)
+   :gender        (get {:m "M" :f "F" :n "N"} gender)
+   :senses        (map (fn [sense] {:id (utils/gen-uuid)  :name (name sense)}) senses)
+   :tenses        (map (fn [tense] {:id (:tense/id tense) :key (name (:tense/key tense)) :value (:tense/value tense)}) tenses)
+   :inflections   (map (fn [inflection] {:id (:inflection/id inflection) :key (name (:inflection/key inflection)) :value (:inflection/value inflection)}) inflections)})
+
+
```
Comment (zmedelis) [inline api/src/api/graphql/translate/dictionary.clj]: Python thing :) Lets stick to idiomatic clojure, one empty line

File: api/src/api/graphql/translate/dictionary.clj [hunk 6]
```diff
@@ -38,3 +39,70 @@
...
+     :phrases      (map (fn [{:keys [language inflections tenses]}]
+                          {:defaultUsage    "YES"
+                           :id              (utils/gen-uuid)
+                           :readerFlagUsage (build-lang-user-flags lang-translation language)
+                           :text            (cond
+                                              (not-empty inflections) (get-default-inflection inflections)
+                                              (not-empty tenses)       (get-default-tense tenses)
+                                              :else "")})
+                        items)
+     :concept      (when (= pos :v)
```
Comment (zmedelis) [inline api/src/api/graphql/translate/dictionary.clj]: This is not used drop the whole `:concept` entry

File: core/src/acc_text/nlg/dictionary/morphology.clj [hunk 7]
```diff
@@ -10,21 +10,19 @@
...
 
-(s/def ::inflection-variants #{:nom-sg :nom-pl :gen-sg :gen-pl})
-
-(s/def ::inflections (s/map-of ::inflection-variants
-                               word?))
+(s/def ::inflections (s/map-of #{:nom-sg :nom-pl :gen-sg :gen-pl} word?))
 
 (s/def ::tenses (s/map-of
-                 #{:present :present-cont :past :imperfect :perfect :future :plu-perfect :future-perfect}
+                 #{:conditional-tense :future-tense :past-tense :present-tense}
```
Comment (zaibacu) [inline core/src/acc_text/nlg/dictionary/morphology.clj]: Do we really need to add `-tense` to each tense?

File: api/src/api/nlg/context.clj [hunk 8]
```diff
@@ -31,12 +31,18 @@
           {}
           (get-values semantic-graph :dictionary-item)))
 
+(defn build-multilang-dictionary-context [semantic-graph _]
```
Comment (zmedelis) [inline api/src/api/nlg/context.clj]: `build-dictionary-context` is still used? If not drop it.

File: api/src/data/entities/dictionary.clj [hunk 9]
```diff
@@ -52,16 +55,41 @@
...
-    (let [{:keys [phrases partOfSpeech name]} (yaml/parse-string (slurp f))
-          filename (utils/get-name f)]
-      (when-not (get-dictionary-item filename)
-        (create-dictionary-item
-          {:key          filename
-           :name         (or name filename)
-           :phrases      phrases
-           :partOfSpeech (when (some? partOfSpeech)
-                           (keyword partOfSpeech))})))))
+  (db/write! reader-flags-db :English  :YES)
```
Comment (zmedelis) [inline api/src/data/entities/dictionary.clj]: At `data.entities.dictionary` we have
```
(defn get-default-flags []
  {:English  :YES
   :Estonian :NO
   :German   :NO
   :Latvian  :NO
   :Russian  :NO})
```
Looks like similar to what you do here.  Any chance of reuse/duplication?

**Pull request merged**